common/cache/cache.go

```diff
@@ -102,6 +105,9 @@ type SimpleOptions struct {
 // deletion, Cache calls go f(i)
 type RemovedFunc func(interface{})
 
+// UpdatedFunc is a type for notifying applications when the cache item was put/deleted
+type UpdatedFunc func(newSize int)
```

Comment:
I guess three things confuse me a little bit when reading the interface.
- The name. I was expecting the function will be called if an item's value was updated, but seems like it's not the case from the implementation. Currently, it's more like a `SizeUpdatedFunc`.
- I think the update function will be more useful if the callback call also gets the new value and probably the key.  `newSize` might be useful but for this particular use case of emitting metrics, ignoring the race condition and get the size using `Size()` method seems also acceptable.
- On item deletion, both `RemovedFunc` and `UpdatedFunc` will be called, we probably should make it clear in the comment in which order are those functions invoked, update first, remove first or maybe in parallel.

Reply:
- The name is bit tricky. Since we are adding this new function callback, I still think we should make the behavior more general/clear in the beginning and not only call the function when size is changed because: 1. it's in the common package and other places can also benefit from it. 2. I don't think that require many code change.
- Regarding parameter, one idea from @yux0 is that we can make it a struct so that it easier to make modifications later.
- Undefined order is probably fine, I just think we need to make it clear in the comments.